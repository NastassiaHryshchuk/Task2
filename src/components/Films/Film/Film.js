import React from 'react';
import classes from './Film.css';

const film = (props) => {
  return (
    <li className={classes.list_item}>
      <div className={classes.flex_container}>
        <span className={classes.list_item_title}>{props.title}</span>
        <span className={classes.list_item_release_date}>{props.releasedate}</span>
      </div>
      <span className={classes.list_item_genre}>{props.genre}</span>
    </li>
  );
};

export default film;
