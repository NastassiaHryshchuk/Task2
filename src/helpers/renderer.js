import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Routes from '../Routes/Routes';

export default(req, store) => {
  const content = renderToString(
    <Provider store={store}>
      <StaticRouter location={req.path} context={{}}>
        <Routes />
      </StaticRouter>
    </Provider>,
  );

  return `
    <!DOCTYPE html>
    <html>
      <head>
        <title>SSR</title>
      </head>
      <body>
        <div id="app">${content}</div>
        <script>
          window.INITIAL_STATE = ${JSON.stringify(store.getState())}
        </script>
        <script src="bundle.js"></script>
      </body>
    </html>
  `;
};
