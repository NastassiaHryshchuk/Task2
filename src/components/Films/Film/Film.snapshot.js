import React from 'react';
import renderer from 'react-test-renderer';

import Film from './Film';

describe('Film Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(
      <Film />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
