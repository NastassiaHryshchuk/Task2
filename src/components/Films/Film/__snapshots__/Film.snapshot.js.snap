// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Snapshot renders 1`] = `
<a
  className="list_item"
  href="#"
>
  <img
    alt={undefined}
    height="600"
    src={undefined}
    width="400"
  />
  <div
    className="flex_container"
  >
    <span
      className="list_item_title"
    />
    <span
      className="list_item_release_date"
    />
  </div>
  <span
    className="list_item_genre"
  />
</a>
`;
